File: backend/internal/model/class.go
package model

import "time"

// Class represents a classroom.
type Class struct {
	ID            string    `json:"id" gorm:"primaryKey"`
	PublicID      string    `json:"publicId" gorm:"uniqueIndex;not null"`
	Name          string    `json:"name" gorm:"not null"`
	StudentCount  int       `json:"studentCount" gorm:"default:0"`
	TotalCapacity int       `json:"totalCapacity" gorm:"default:30"`
	IsActive      bool      `json:"isActive" gorm:"default:true"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

File: frontend/src/types/class.ts
export interface ClassData {
  publicId: string;
  name: string;
  joinLink: string;
  qrCodeUrl?: string;
}

export interface ClassInfo {
  publicId: string; // Use publicId as the main identifier
  name: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface ClassResponse {
  class: ClassInfo;
  joinLink: string;
}

export interface ClassSeatMap {
  [seatNumber: number]: {
    studentId?: number;
    studentName: string;
    isGuest: boolean;
    isEmpty: boolean;
    score: number;
  };
}

export interface ClassWithSeatMap extends ClassInfo {
  seatMap: ClassSeatMap;
  totalCapacity: number;
  availableSlots: number;
  initialized?: boolean; // Use this flag to track initialization
}